src/ethereum/Contract.ts

```diff
@@ -86,40 +86,68 @@ export abstract class Contract<T = any> {
 
   /**
    * Execute a write-operation, and broadcast it to be processed by miners and published on the blockchain.
-   * @param  {string}    method - Method name
-   * @param  {...object} args   - Every argument after the name will be fordwarded to the transaction method, in order
+   * @param  {string} method - Method name
+   * @param  {string} args - Args of the method type name. E.g: address,address,uint256
+   * @param  {...object} params   - Every argument after the name will be fordwarded to the call function, in order
    * @return {Promise} - promise that resolves when the transaction does
    */
-  sendTransaction(method: string, ...args) {
+  sendTransactionByType(method: string, args?: string, ...params) {
+    const fn = this.instance[method][args]
+    return this._sendTransaction(fn, method, ...params)
+  }
+
+  /**
+   * Inoke a contract function that does not broadcast or publish anything on the blockchain.
+   * @param  {string} method - Method name
+   * @param  {string} args - Args of the method type name. E.g: address,address,uint256
+   * @param  {...object} params   - Every argument after the name will be fordwarded to the call function, in order
+   * @return {Promise} - promise that resolves when the call does
+   */
+  sendCallByType(method: string, args?: string, ...params) {
+    const fn = args ? this.instance[method][args] : this.instance[method]
+    return this._sendCall(fn, method, ...params)
+  }
```

Comment:
```ts
sendCallByType(method: string, args?: string, ...params)
```
This signature can be better, what happens in the following scenarios?
```ts
sendCallByType('method','uint256',1)
sendCallByType('method',1)
```

The first one is correct.
The second one won't work, because `args` is 1
```ts
const fn = args ? this.instance[method][args] : this.instance[method]
```
and it will try to find a function 
```ts
this.instance['method'][1]
```

Reply:
It is what we want. If the user calls it wrong without specifying a valid method will throw.

Reply:
What about adding a doc string setting the parameter as optional?

I am ok with both, but it would be great to define.